### DISCUSSION TITLE
WinterCG fetch handler

### DISCUSSION THREAD

**Discussion by xantiagoma** [remix-run/remix#9220] [Proposals]
Hi Guys,

It would be great if Remix has a WinterCG compatible _fetchHandler_ method `remixApp.fetch(request: Resquet): Response` or as an utility in a new package, similar to:

- GraphQL Yoga [server.handleRequest](https://the-guild.dev/graphql/yoga-server/docs/integrations/integration-with-nextjs) / [server.fetch](https://the-guild.dev/graphql/yoga-server/docs/integrations/z-other-environments)
- TRPC [fetchRequestHandler](https://trpc.io/docs/server/adapters/fetch)
- ElysiaJS [handle](https://elysiajs.com/patterns/unit-test.html#unit-test) / [fetch](https://elysiajs.com/integrations/sveltekit.html)
- HonoJS [fetch](https://hono.dev/api/hono#fetch) / [request](https://hono.dev/api/hono#request)

I saw there's already one for express / node http-server using `createRequestHandler`
```ts
const { createRequestHandler } = require("@remix-run/express");
```
as is used here https://sergiodxa.com/tutorials/run-next-and-remix-on-the-same-server, having the fetch handler would make possible to integrate Remix as sub-routes of Elysia / Hono / Hattip / any WinterCG server

Examples:

```ts
Bun.serve({
  fetch(req, server) {
   return remixApp.fetch(req);
   },
})
```

```ts
new Elysia().all('/**', async ({request}) => remixApp.fetch(request) )
```

```ts
const app = new Hono()
app.get('/**', (c) => remixApp.fetch(c.request))
```

Also It could be used for testing purposes similar to what is shown on [ElysiaJS Testing](https://elysiajs.com/patterns/unit-test.html#unit-test)

**Comment by sergiodxa** [comment]
The `@remix-run/server-runtime` package has a createRequestHandler function that works on any runtime.

This one is used by the Express package to run as a route in Express, also by the Cloudflare Pages package, and other official packages.

This is because Remix is designed to be a request handler in any http server, and it’s what enables things like remix-hono which is an adapter to run Remix as a route in Hono

It’s also possible to use it with Bun as you can read more in https://sergiodxa.com/tutorials/build-an-app-with-remix-and-bun.

**Reply by xantiagoma** [reply]
That's great, thank you for the link I swear I google it for minutes and couldn’t got info 😅.
